Content re-organization

lucperkins opened the pull request.

Hoverbear commented on layouts/partials/navbar.html:
```diff
@@ -45,11 +46,31 @@
+        </div>
+
+        <div class="navbar-item has-dropdown is-hoverable">
+          <a class="navbar-link" href="/docs">
```
```suggestion
          <a class="navbar-link" href="/deep-dive/">
```

Hoverbear commented on layouts/partials/navbar.html:
```diff
@@ -45,11 +46,31 @@
               {{ .Title }}
             </a>
-            {{- end }} <!-- range $docs -->
+            {{ end }} <!-- range $docs -->
```
Right now the `Docs` ion Navbar links to `/docs/` (a blank page), can it go to `/docs/concepts/overview/`?

lucperkins replied:
Good catch

lucperkins commented:
@Hoverbear I think I've addressed your concerns. I've also added a redirect for `/docs -> /docs/concepts/overview` in case there are any stray links out there.
